### PR TITLE
ICU-21073 Fix ICU Data Build Tool in "locale" mode.

### PR DESCRIPTION
**Pull request by sffc**
##### Checklist

- [x] Issue filed: https://unicode-org.atlassian.net/browse/ICU-21073
- [x] Updated PR title and link in previous line to include Issue number
- [x] Issue accepted
- [x] Tests included
- [x] Documentation is changed or added

### CODE REVIEW

**Comment by jefgen** [comment]
This should probably go into `maint/maint-67`, right?

File: icu4c/source/data/BUILDRULES.py [hunk 1]
```diff
@@ -399,6 +400,29 @@ def generate_curr_supplemental(config, io, common_vars):
...
+    input_basename = "tzdbNames.txt"
+    output_file = OutFile("zone/tzdbNames.res")
+    return [
+        SingleExecutionRequest(
+            name = "zone_supplemental_res",
+            category = "zone_supplemental",
+            dep_targets = [],
+            input_files = [input_file],
+            output_files = [output_file],
+            tool = IcuTool("genrb"),
```
Comment (jefgen) [inline icu4c/source/data/BUILDRULES.py:415]:
```
2020-04-14T00:57:47.4906343Z LD_LIBRARY_PATH=../lib:../stubdata:../tools/ctestfw:$LD_LIBRARY_PATH  ../bin/genrb -s ./out/tmp/in/zone -d ./out/build/icudt67l/zone -i ./out/build/icudt67l -k tzdbNames.txt
```
👍

**Comment by sffc** [comment]
> This should probably go into `maint/maint-67`, right?

Thanks!  Fixed.

**Comment by sffc** [comment]
The AppVeyor failure is:

```
make[2]: Entering directory '/cygdrive/c/projects/icu/icu4c/source/tools/toolutil'
ln -fs libicutu67.dll.a ../../lib/libicutu.dll.a
ln: failed to create symbolic link '../../lib/libicutu.dll.a': File exists
make[2]: Leaving directory '/cygdrive/c/projects/icu/icu4c/source/tools/toolutil'
```

I will ignore and submit despite the error.

**Pull request merged**